corehq/apps/repeaters/models.py

```diff
@@ -334,7 +376,10 @@ def get_url(self, repeat_record):
             # adapted from http://stackoverflow.com/a/2506477/10840
             url_parts = list(urlparse.urlparse(url))
             query = urlparse.parse_qsl(url_parts[4])
-            query.append(("app_id", self.payload_doc(repeat_record).app_id))
+            try:
+                query.append(("app_id", self.payload_doc(repeat_record).app_id))
+            except (XFormNotFound, ResourceNotFound):
+                return None
```

Comment:
This is intended to match exactly previous behavior. It was moved in from `Repeater.url` in https://github.com/dimagi/commcare-hq/pull/16059/commits/12b6288bfc7a3c656aac9b0c1515f2f0a1f3f6e4. I also checked that this function (`get_url`) was not called anywhere other than in `Repeater.url`, (and now after this change, it's not called anywhere except from places that _used_ to call the now-deprecated `Repeater.url`).

Why it was originally chosen to catch these particular errors and return `None`, I didn't try to deeply re-examine. On the face of it, it's a reasonable contract for `get_url` not to expose whatever the underlying errors are and just return `None` to mean "I couldn't get that", but of course the caller would need to check for that, and as you probably guessed, they don't. This again, just makes that clearer but doesn't try to fix it.